File: backup/LineBotConfiguration.java
//package org.occidere.githubnotifier.configuration;
//
//import com.linecorp.bot.client.LineMessagingClient;
//import org.springframework.beans.factory.annotation.Value;
//import org.springframework.context.annotation.Bean;
//import org.springframework.context.annotation.Configuration;
//
///**
// * @author occidere
// * @Blog: https://occidere.blog.me
// * @Github: https://github.com/occidere
// * @since 2020-04-01
// */
//@Configuration
//public class LineBotConfiguration {
//
//    @Bean
//    public LineMessagingClient lineMessagingClient(@Value("${line.channel.token}") String lineChannel) {
//        return LineMessagingClient.builder(lineChannel).build();
//    }
//}

File: src/main/scala/org/occidere/githubnotifier/configuration/LineBotConfiguration.scala
package org.occidere.githubnotifier.configuration

import java.util.Properties

import com.linecorp.bot.client.LineMessagingClient
import org.springframework.context.annotation.{Bean, Configuration}
import org.springframework.core.io.ClassPathResource
import org.springframework.core.io.support.PropertiesLoaderUtils

/**
 * @author occidere
 * @Blog: https://blog.naver.com/occidere
 * @Github: https://github.com/occidere
 * @since 2020. 08. 17.
 */
@Configuration
class LineBotConfiguration {
  private val props: Properties = PropertiesLoaderUtils.loadProperties(new ClassPathResource("/application.properties"))

  private val lineChannel: String = props.getProperty("line.channel.token")

  @Bean
  def lineMessagingClient: LineMessagingClient = LineMessagingClient.builder(lineChannel).build
}
